evm/utils/blobs.py

```diff
@@ -0,0 +1,68 @@
+from itertools import (
+    chain,
+)
+
+from eth_utils import (
+    keccak,
+)
+
+from evm.constants import (
+    CHUNK_SIZE,
+    COLLATION_SIZE,
+)
+
+
+def chunk_iterator(blob):
```

Comment:
The **blob serialization** process would be more complicated than partitions. It requires some special encoding to serialize blobs to chunks. I guess Vitalik or Justin will post some preliminary ideas on ethresear.ch soon. Right now you can mark a `TODO: blob serialization` comment.

Reply:
I think I've got the terminology wrong: What I called `blob` should actually be `collation_body`. My (updated) understanding right now:

- `collation_body`: `2**20` bytes of binary data
- `chunks`: 32 bytes piece of the body (so the body is made up of 32768 chunks, simply stringed together), needed to calculate the chunks root
- `blobs`: Another "interpretation" of the collation body. Eventually these will be transactions and we'll need some way to press blobs into the chunk structure, but for phase 1 we can don't need to consider them at all.

Is this correct?